pkg/controller/rds/dbparametergroup/setup.go

```diff
@@ -0,0 +1,150 @@
+// 	if err != nil {
+// 		return managed.ExternalCreation{}, err
+// 	}
+// 	meta.SetExternalName(cr, awsclients.StringValue(obj.DBParameterGroup.DBParameterGroupName))
+// 	return managed.ExternalCreation{ExternalNameAssigned: true}, nil
+// }
+
+func preCreate(_ context.Context, cr *svcapitypes.DBParameterGroup, obj *svcsdk.CreateDBParameterGroupInput) error {
+	obj.DBParameterGroupName = awsclients.String(meta.GetExternalName(cr))
+
+	return nil
+}
+
+func preUpdate(_ context.Context, cr *svcapitypes.DBParameterGroup, obj *svcsdk.ModifyDBParameterGroupInput) error {
+	obj.DBParameterGroupName = awsclients.String(meta.GetExternalName(cr))
+	obj.Parameters = make([]*rds.Parameter, len(cr.Spec.ForProvider.Parameters))
+
+	for i, v := range cr.Spec.ForProvider.Parameters {
+		// check if mandatory parameters are set (ApplyMethod, ParameterName, ParameterValue)
+		if (v.ApplyMethod == nil) || (v.ParameterName == nil) || (v.ParameterValue == nil) {
+			return errors.New("ApplyMethod, ParameterName and ParameterValue are mandatory fields and can not be nil")
+		}
+		obj.Parameters[i] = &rds.Parameter{
+			ApplyMethod:    awsclients.String(*v.ApplyMethod),
+			ParameterName:  awsclients.String(*v.ParameterName),
+			ParameterValue: awsclients.String(*v.ParameterValue),
+		}
+	}
+	return nil
+}
+
+func preDelete(_ context.Context, cr *svcapitypes.DBParameterGroup, obj *svcsdk.DeleteDBParameterGroupInput) (bool, error) {
+	obj.DBParameterGroupName = awsclients.String(meta.GetExternalName(cr))
+	return false, nil
+}
+
+func (e *custom) isUpToDate(cr *svcapitypes.DBParameterGroup, obj *svcsdk.DescribeDBParameterGroupsOutput) (bool, error) {
+	// TODO(Dkaykay): We need isUpToDate to have context.
+	ctx := context.TODO()
+	results, err := e.getCurrentDBParameters(ctx, cr)
+
+	// compare CR with currently set Parameters
+	for _, v := range cr.Spec.ForProvider.Parameters {
+		for _, w := range results {
+			if *v.ParameterName == *w.ParameterName {
+				switch {
+				case (v.ParameterValue == nil) || (w.ParameterValue == nil):
+					return false, nil
+				case (v.ParameterValue == nil) && (w.ParameterValue == nil):
+					return true, nil
+				case (*v.ParameterValue != *w.ParameterValue) || (*v.ApplyMethod != *w.ApplyMethod):
+					return false, nil
+				}
+			}
+		}
+	}
+	return true, err
+}
+
+func (e *custom) getCurrentDBParameters(ctx context.Context, cr *svcapitypes.DBParameterGroup) ([]*svcsdk.Parameter, error) {
+	input := &rds.DescribeDBParametersInput{
+		DBParameterGroupName: awsclients.String(meta.GetExternalName(cr)),
+		MaxRecords:           awsclients.Int64(20),
+	}
+	pageNum := 0
+	var results []*svcsdk.Parameter
+	err := e.client.DescribeDBParametersPagesWithContext(ctx, input, func(page *rds.DescribeDBParametersOutput, lastPage bool) bool {
+		pageNum++
+		results = append(results, page.Parameters...)
+		return pageNum <= 20
```

Comment:
```suggestion
		results = append(results, page.Parameters...)
		return !lastPage
```

From my reading of the API, we need to return `false` if we'd like to stop and it seems `!lastPage` would be suitable for this. Is there a specific reason for stopping at page 20?

Reply:
changed following suggestion